base/compiler/typeinfer.jl

```diff
@@ -822,6 +822,8 @@ function typeinf_edge(interp::AbstractInterpreter, method::Method, @nospecialize
             unlock_mi_inference(interp, mi)
             return Any, nothing
         end
+        caller_kwfunc_inlining = caller.kwfunc_inlining
+        caller_kwfunc_inlining !== nothing && propagate_caller_annotations!(caller_kwfunc_inlining, frame)
```

Comment:
hm, wait, can't there be any number of callers, and this is just the first one that we saw, before we put it in the cache?